Runtime/Tracking/Modification/TransformPropertyApplier.cs

```diff
@@ -152,8 +152,7 @@ public class UnityEvent : UnityEvent<EventData>
         /// <param name="source">The data to build the new source from.</param>
         public virtual void SetSource(GameObject source)
         {
-            sourceTransformData.Clear();
-            sourceTransformData.Transform = source != null ? source.transform : null;
+            Source = source != null ? new TransformData(source) : null;

```

Comment:
As mentioned in slack. I just wonder if trying to avoid GC here is going against the normal operation of setting a new source.

If you do this to set the Source via code:

```
TransformData myData = new TransformData();
applier.Source = myData;
```

Then you're effectively doing the same as the `SetSource(GameObejct)` method.